File: reverse-integer.py
# class Solution(object):
#     def reverse(self, x):
#         """
#         :type x: int
#         :rtype: int
#         """
#         rev = int(str(abs(x))[::-1])
#         return 0 if len(str(bin(rev))[2:]) > 31 else int(rev) if x > 0 else -int(rev)

class Solution(object):
    def get_rev(self, n):
        temp = 0
        while n:
            rem = n%10
            n /=10
            temp = temp*10 + rem
        if temp > 2**31:
            temp = 0
        return temp

    def reverse(self, x):
        """
        :type x: int
        :rtype: int
        """
        return self.get_rev(x) if x >= 0 else -self.get_rev(-x)

File: reverse-integer.cpp
class Solution {
public:
    int reverse(int x) {
        int temp = x;
        vector<int> rev ;
        vector<int> rev_bin;
        long create_int = 0;
        long val = 1;
        
        x = abs(x);
        while(x){
        	rev.push_back(x%10);
        	x = x/10;
        }
        int size = rev.size();
        for(int i = 0;i< size;i++){
        	create_int = create_int + rev[size - 1 -i]*val;
        	val = val*10;
        }

        long t = create_int;
        while(t){
        	rev_bin.push_back(t%2);
        	t = t/2;
        }

        
        if (rev_bin.size() > 31)
        	return 0;
        else{
        	if (temp > 0)
        		return int(create_int);
        	else
        		return -int(create_int);
        }
        
    }
};